javaslang/src/main/java/javaslang/Value.java

```diff
@@ -954,12 +954,15 @@ default CharSeq toCharSeq() {
      */
     @SuppressWarnings("unchecked")
     default PriorityQueue<T> toPriorityQueue() {
+        final Comparator<T> comparator;
         if (this instanceof PriorityQueue<?>) {
             return (PriorityQueue<T>) this;
+        } else if (this instanceof Ordered<?>) {
+            comparator = ((Ordered<T>) this).comparator();
         } else {
-            final Comparator<T> comparator = (Comparator<T> & Serializable) (o1, o2) -> ((Comparable<T>) o1).compareTo(o2);
-            return toPriorityQueue(comparator);
+            comparator = (Comparator<T> & Serializable) (o1, o2) -> ((Comparable<T>) o1).compareTo(o2);
         }
+        return toPriorityQueue(comparator);
```

Comment:
The control-flow is somehow 'overlapping' because the if-else-block cannot be substituted with a function call.

Please use this more 'straight-forward' version:

```java
if (this instanceof PriorityQueue) {
    return (PriorityQueue<T>) this;
} else {
    final Comparator<T> comparator = (this instanceof Comparator)
        ? ((Comparator<T>) this)
        : (Comparator<T> & Serializable) (o1, o2) -> ((Comparable<T>) o1).compareTo(o2);
    return toPriorityQueue(comparator);
}
```